src/Chapter3.hs

```diff
@@ -541,34 +539,53 @@ newtype Houses = Houses
 
 data MagicalCity = MagicalCity
     {
-      castle :: Castle,
+      castle :: Maybe Castle,
       building :: Building,
       houses :: [Houses]
     }
 
-
-buildCastle :: String -> MagicalCity -> MagicalCity
-buildCastle newname city = city { castle = newCastle }
+buildCastle :: String -> Maybe MagicalCity -> MagicalCity
+buildCastle newname _ = createCity
   where
-    ccity = castle city
-    newCastle = ccity { __name__ = newname }
-
+    newCastle = Castle { __name__ = newname, wall = False }
+    newHouses = [Houses { __people__ = 1 }] 
+    createCity = MagicalCity { 
+      castle = Just newCastle, 
+      building = Library, 
+      houses = newHouses
```

Comment:
This is strange that you erase all previous city data and write new default 🤔

Reply:
buildCastle — build a castle in the city. If the city `already has a castle, the old castle is destroyed`, and the new castle with the __new name__ is built

I misunderstanding the exercise ?

Reply:
Ohh I got it ! Sorry I destroy everything x) 
Ok I change that